src/runtime/runtime_mimxrt1062_clock.go

```diff
@@ -0,0 +1,1242 @@
+}
+
+func enableTimerClocks() {
+
+	// ---------------------------------------------------------------- GPT/PIT --
+	clkIpGpt1.enable(true)  // enable GPT/PIT clock gates
+	clkIpGpt1S.enable(true) //  ~
+	clkIpGpt2.enable(true)  //  ~
+	clkIpGpt2S.enable(true) //  ~
+	clkIpPit.enable(true)   //  ~
+}
+
+func enablePinClocks() {
+
+	// ----------------------------------------------------------------- IOMUXC --
+	clkIpIomuxcGpr.enable(true) // enable IOMUXC clock gates
+	clkIpIomuxc.enable(true)    //  ~
+
+	// ------------------------------------------------------------------- GPIO --
+	clkIpGpio1.enable(true) // enable GPIO clock gates
+	clkIpGpio2.enable(true) //  ~
+	clkIpGpio3.enable(true) //  ~
+	clkIpGpio4.enable(true) //  ~
+}
+
+// enableClocks enables the clock gates for peripherals used by this runtime.
+func enablePeripheralClocks() {
+
+	// -------------------------------------------------------------------- ADC --
+	clkIpAdc1.enable(true) // enable ADC clock gates
+	clkIpAdc2.enable(true) //  ~
+
+	// ------------------------------------------------------------------- XBAR --
+	clkIpXbar1.enable(true) // enable XBAR clock gates
+	clkIpXbar2.enable(true) //  ~
+	clkIpXbar3.enable(true) //  ~
+
+	// ------------------------------------------------------------------ USDHC --
+	clkIpUsdhc1.enable(true) // enable USDHC clock gates
+	clkIpUsdhc2.enable(true) //  ~
+
+	// ------------------------------------------------------------------- SEMC --
+	clkIpSemc.enable(true) // enable SEMC clock gate
+
+	// ---------------------------------------------------------------- FLEXSPI --
+	clkIpFlexSpi2.enable(true) // enable FLEXSPI2 clock gate
+
+	// ------------------------------------------------------------------ LPSPI --
+	clkIpLpspi1.enable(true) // enable LPSPI clock gate
+	clkIpLpspi2.enable(true) //  ~
+	clkIpLpspi3.enable(true) //  ~
+	clkIpLpspi4.enable(true) //  ~
+
+	// ------------------------------------------------------------------ LPI2C --
+	clkIpLpi2c1.enable(true) // enable LPI2C clock gate
+	clkIpLpi2c2.enable(true) //  ~
+	clkIpLpi2c3.enable(true) //  ~
+
+	// -------------------------------------------------------------------- CAN --
+	clkIpCan1.enable(true)  // enable CAN clock gate
+	clkIpCan2.enable(true)  //  ~
+	clkIpCan3.enable(true)  //  ~
+	clkIpCan1S.enable(true) //  ~
+	clkIpCan2S.enable(true) //  ~
+	clkIpCan3S.enable(true) //  ~
+
+	// ------------------------------------------------------------------- UART --
+	clkIpLpuart1.enable(true) // enable UART clock gate
+	clkIpLpuart2.enable(true) //  ~
+	clkIpLpuart3.enable(true) //  ~
+	clkIpLpuart4.enable(true) //  ~
+	clkIpLpuart5.enable(true) //  ~
+	clkIpLpuart6.enable(true) //  ~
+	clkIpLpuart7.enable(true) //  ~
+	clkIpLpuart8.enable(true) //  ~
+
+	// ----------------------------------------------------------------- FLEXIO --
+	clkIpFlexio1.enable(true) // enable FLEXIO clock gates
+	clkIpFlexio2.enable(true) //  ~
+}
+
+type clock uint32
+
+// Named oscillators
```

Comment:
Also consider adding to a `device` package file.

Reply:
Moved to new file `src/device/nxp/clock_mimxrt1062.go`